dev/benchmarks/macrobenchmarks/pubspec.yaml

```diff
@@ -15,15 +15,8 @@ dependencies:
   #   //packages/flutter_tools/lib/src/commands/update_packages.dart
   # and run
   #   flutter update-packages --force-upgrade
-  flutter_gallery_assets: 0.2.4
+  flutter_gallery_assets: 0.2.5
 
-  # This is needed for web_benchmarks.
-  gallery:
-    git:
-      url: https://github.com/flutter/gallery.git
```

Comment:
@pennzht @yjbanov I'm unfortunately removing the flutter/flutter <-> flutter/gallery linkage here. 

The reason being that flutter/flutter can no longer proactively control the pub dependencies it uses (since they're effectively controlled by flutter/gallery with the one version rule) and increases maintenance cost. 

I've been trying to cycle package updates through flutter/gallery first as well but have trouble rolling to gallery head and haven't been able to resolve it this week https://github.com/flutter/gallery/pull/282#issuecomment-686859738 to unblock Flutter's 1.22 branch cut. 

Some context: https://github.com/flutter/gallery/issues/254#issuecomment-686697714, https://github.com/flutter/gallery/issues/254#issuecomment-686812713. 

Could web benchmarks be driven off of flutter/flutter's gallery https://github.com/flutter/flutter/tree/master/dev/integration_tests/flutter_gallery instead of the flutter/gallery gallery?

Is this acceptable?

Reply:
> Is this acceptable?

That would be very unfortunate. We were going to expand usage of the new gallery to benchmark the CanvasKit renderer as well. This will halt that effort too.

We have similar linkage between flutter/flutter and flutter/plugins. What are we doing wrong with flutter/gallery that it leads to so much pain?

Reply:
Could you point me to it? I didn't find it in the codebase.

Reply:
The `macrobenchmarks` app depends on several plugins, such as [shared_preferences](https://github.com/flutter/flutter/blob/e641195aa6666df26518d69c4f17271501642de1/dev/benchmarks/macrobenchmarks/pubspec.yaml#L62) and `url_launcher`.

Reply:
Ah I see what you mean. Right it does. flutter/flutter has pub dependencies against released pub packages which is fine. Pub can dependency resolve the asks in pubspec.yaml for all pubspecs in the repo and produce a pubspec.lock (which seems to be fed back into pubspec.yaml in flutter/flutter's case). It doesn't add a git dependency on the flutter/plugin repo. 

The difference here is that it's no longer a 1-tiered resolution. The pubspec.lock resolution in the gallery repo is already resolved. Those become the specific pubspec.yaml asks feeding into flutter/flutter. 

If I change a shared_preferences dependency in flutter/flutter today, I can use flutter update-packages --force-upgrade. It'll find all instances and pull a valid resolved version. It can't account for flutter/gallery.

This is of course solvable, but it implies building more tooling into flutter/flutter to account for flutter/gallery and push synchronized PRs back into flutter/gallery.

Reply:
@pennzht has a plan to fix this (https://docs.google.com/document/d/1eI9OqjGbyKSAGQVupN6Wjz3NmrOjhLG2JSPWkvJMyfs). In the meantime, is it possible to make it work with the current setup, so we have time to fix it in the next couple of weeks?

Reply:
Thanks for taking this on. Yes, I can wait. I've done a reduced-scope workaround for the time being. My timing concerns are mainly around getting our visuals correct now that iOS 14 is launched so I'd like to get this through at some point this month but it's not urgent.

Reply:
Ok, so let's remove the gallery benchmarks for now. After a discussion with @pennzht @Hixie @jmagman @jonahwilliams and @christopherfujino we decided to move the benchmarks into the customer shard. We'll start working on that.